lib/src/interceptor.dart

```diff
@@ -13,16 +13,30 @@ final registry = Registry();
 
 class Registry {
   final _interceptors = <Interceptor>[];
+  final _automaticInterceptors = <Interceptor>[];
 
-  void cleanAll() => _interceptors.clear();
+  void cleanAll() {
+    _interceptors.clear();
+    _automaticInterceptors.clear();
+  }
 
   Iterable<Interceptor> get pendingMocks => _interceptors.where(
         (interceptor) => !interceptor.isDone,
       );
 
   Iterable<Interceptor> get activeMocks => _interceptors;
 
-  void add(Interceptor interceptor) => _interceptors.add(interceptor);
+  void add(Interceptor interceptor) {
+    _interceptors.add(interceptor);
+
+    if (interceptor.generateAuto){
+      if (interceptor._matcher.method.toUpperCase() == 'GET') {
```

Comment:
I apologise for this.  I think the correct way to handle this is to create a separate subclass for each method type (and throwing clause), and put this kind of behaviour in there.  But that seemed like a fairly major piece of refactoring, and I didn't want to do that without your approval.  But, I wanted you to know, I wouldn't have done this in my code :-)